### PR TITLE
DB connection done.

### PR DESCRIPTION
**Pull request by Boorhan**
**Description:**
We are going to use MySQL for our database. We need to make sure our database connection is established or not.

We have to create a database in MySQL and connect it to our project for that we need to configure.

**Configuration:**
We use .env files to manage environment-specific configuration properties.
Steps:

1. Add the dotenv Dependency

2. Create the .env File

3. Add the dotenv Dependency

4. Reference Environment Variables in application.yml

**Check Connectivity:** 
Run this command:

`curl http://localhost:8080/actuator/health`

Auto generated File Issue: 
It's not considered good practice to keep auto-generated files in GitHub. Because of Redundancy, Increase repository size, Merge conflict issue and also Inconsistent builds.

**Trello link:**
https://trello.com/c/6bnM3aRp/14-003-db-connection

### CODE REVIEW

File: src/main/java/com/demo/classroom/ClassroomApplication.java [hunk 1]
```diff
@@ -3,10 +3,19 @@
...
 public class ClassroomApplication {
 
 	public static void main(String[] args) {
+		
+		Dotenv dotenv = Dotenv.load();
+        System.setProperty("spring.datasource.url", dotenv.get("SPRING_DATASOURCE_URL"));
+        System.setProperty("spring.datasource.username", dotenv.get("SPRING_DATASOURCE_USERNAME"));
+        System.setProperty("spring.datasource.password", dotenv.get("SPRING_DATASOURCE_PASSWORD"));
+
+
```
Comment (rummansynergy) [inline src/main/java/com/demo/classroom/ClassroomApplication.java]: fix the formatting.

**Review by rummansynergy — CHANGES_REQUESTED** [review]

File: .env.example [hunk 2]
```diff
@@ -0,0 +1,3 @@
+SPRING_DATASOURCE_URL=jdbc:mysql://localhost:3306/your_database_name
+SPRING_DATASOURCE_USERNAME=your_username
+SPRING_DATASOURCE_PASSWORD=your_password
```
Comment (rummansynergy) [inline .env.example]:
```suggestion
DB_HOST = jdbc:mysql://localhost:3306/
DB_NAME = your_database_name
DB_USERNAME = your_username
DB_PASSWORD = your_password
```

File: src/main/java/com/demo/classroom/ClassroomApplication.java [hunk 3]
```diff
@@ -3,10 +3,19 @@
...
 @SpringBootApplication
 public class ClassroomApplication {
 
 	public static void main(String[] args) {
+		
+		Dotenv dotenv = Dotenv.load();
+        System.setProperty("spring.datasource.url", dotenv.get("SPRING_DATASOURCE_URL"));
+        System.setProperty("spring.datasource.username", dotenv.get("SPRING_DATASOURCE_USERNAME"));
+        System.setProperty("spring.datasource.password", dotenv.get("SPRING_DATASOURCE_PASSWORD"));
```
Comment (rummansynergy) [inline src/main/java/com/demo/classroom/ClassroomApplication.java]: Is there any way to directly assign this values to yml?

**Review by rummansynergy — CHANGES_REQUESTED** [review]
Add story link to the description.

File: src/main/java/com/demo/classroom/ClassroomApplication.java (hunk 1 above)
Comment (Boorhan) [reply src/main/java/com/demo/classroom/ClassroomApplication.java]: Solved.

File: .env.example (hunk 2 above)
Comment (Boorhan) [reply .env.example]: Solved and done.

File: src/main/java/com/demo/classroom/ClassroomApplication.java (hunk 3 above)
Comment (Boorhan) [reply src/main/java/com/demo/classroom/ClassroomApplication.java]: Done.

**Review by rummansynergy — APPROVED** [review]

**Pull request merged**